libs/ledger-live-common/src/hw/ftsFetchImage.ts

```diff
@@ -0,0 +1,204 @@
+import { Observable, from, of, throwError } from "rxjs";
+import { catchError, concatMap, delay, mergeMap } from "rxjs/operators";
+import {
+  DeviceOnDashboardExpected,
+  TransportError,
+  TransportStatusError,
+} from "@ledgerhq/errors";
+import { ungzip } from "pako";
+
+import { withDevice } from "./deviceAccess";
+import getDeviceInfo from "./getDeviceInfo";
+import getAppAndVersion from "./getAppAndVersion";
+import { isDashboardName } from "./isDashboardName";
+import attemptToQuitApp, { AttemptToQuitAppEvent } from "./attemptToQuitApp";
+
+const MAX_APDU_SIZE = 240;
+
+export type FetchImageEvent =
+  | AttemptToQuitAppEvent
+  | {
+      type: "progress";
+      progress: number;
+    }
+  | {
+      type: "imageFetched";
+      hexImage: string;
+    };
+
+export type FetchImageRequest = {
+  deviceId: string;
+};
+
+export default function fetchImage({
+  deviceId,
+}: FetchImageRequest): Observable<FetchImageEvent> {
+  const sub = withDevice(deviceId)(
+    (transport) =>
+      new Observable((subscriber) => {
+        const timeoutSub = of<FetchImageEvent>({
+          type: "unresponsiveDevice",
+        })
+          .pipe(delay(1000))
+          .subscribe((e) => subscriber.next(e));
```

Comment:
Did you try to run this action with a locked device ? What error do you get ? `5515`? If yes - i don't think we need to fix it in this PR, but we should add a ticket